src/test/java/com/mongodb/kafka/connect/sink/cdc/qlik/rdbms/RdbmsHandlerTest.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.kafka.connect.sink.cdc.qlik.rdbms;
+
+import static com.mongodb.kafka.connect.sink.SinkTestHelper.createTopicConfig;
+import static java.util.Collections.emptyMap;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.DynamicTest.dynamicTest;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.kafka.connect.errors.DataException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import org.bson.BsonDocument;
+
+import com.mongodb.client.model.DeleteOneModel;
+import com.mongodb.client.model.ReplaceOneModel;
+import com.mongodb.client.model.UpdateOneModel;
+import com.mongodb.client.model.WriteModel;
+
+import com.mongodb.kafka.connect.sink.cdc.qlik.OperationType;
+import com.mongodb.kafka.connect.sink.converter.SinkDocument;
+
+@RunWith(JUnitPlatform.class)
+class RdbmsHandlerTest {
```

Comment:
I see you handle the Qlik envelope in the code, but there are no tests against the case of no qlik envelope.

Reply:
Hi @abraham-leal 

I think all operation parsing should be covered by the OperationHandlerTest. This test just covers a single type of message so not to duplicate the testing.

If I'm not testing something I should could you provide the json of a sample record so I can investigate?

Reply:
I see only tests for the object created from the OP in `OperationTypeTest`, what I mean is testing for the parsing itself. There can be two types of records by Qlik: With and Without envelope.

With envelope: "{message: { data: {id: 1234, foo: 'bar'}, headers: { operation: 'INSERT'}}}"
Without envelope: "{ data: {id: 1234, foo: 'bar'}, headers: { operation: 'READ'}}"

I see `getSubDocumentOrOriginal` can handle this case, I just don't see it being tested either in `OperationHelper` or `RdbmsHandler`.

Reply:
Perfect - we ensure I add tests for that scenario.

Thanks again @abraham-leal